openpype/hosts/equalizer/api/host.py

```diff
@@ -0,0 +1,188 @@
+"""3dequalizer host implementation.
+
+note:
+    3dequalizer 7.1v2 uses Python 3.7.9
+
+"""
+import json
+import os
+import re
+
+import pyblish.api
+import tde4  # noqa: F401
+from attrs import asdict
+from attrs.exceptions import NotAnAttrsClassError
+from qtpy import QtCore, QtWidgets
+
+from openpype.host import HostBase, ILoadHost, IPublishHost, IWorkfileHost
+from openpype.hosts.equalizer import EQUALIZER_HOST_DIR
+from openpype.hosts.equalizer.api.pipeline import Container
+from openpype.pipeline import (
+    register_creator_plugin_path,
+    register_loader_plugin_path,
+)
+from openpype.tools.utils import get_openpype_qt_app
+
+CONTEXT_REGEX = re.compile(
+    r"AYON_CONTEXT::(?P<context>.*?)::AYON_CONTEXT_END",
+    re.DOTALL)
+PLUGINS_DIR = os.path.join(EQUALIZER_HOST_DIR, "plugins")
+PUBLISH_PATH = os.path.join(PLUGINS_DIR, "publish")
+LOAD_PATH = os.path.join(PLUGINS_DIR, "load")
+CREATE_PATH = os.path.join(PLUGINS_DIR, "create")
+INVENTORY_PATH = os.path.join(PLUGINS_DIR, "inventory")
+
+
+class EqualizerHost(HostBase, IWorkfileHost, ILoadHost, IPublishHost):
+    name = "equalizer"
+    _instance = None
+
+    def __new__(cls):
+        if not hasattr(cls, "_instance") or not cls._instance:
+            cls._instance = super(EqualizerHost, cls).__new__(cls)
+        return cls._instance
```

Comment:
Why is this needed? I don't see other hosts implementations doing this?

If there are certain reasons why this is required for 3DEqualizer it might be good to add a comment as to WHY that's the case.

Reply:
This is just to be sure that host is a singleton. I don't think we enforce it anywhere upstream and I am not sure what exact side-effect there could be if multiple host objects are created, but because of rather fragile way is is using callback timer and storing json encoded context on project notes, I'd rather be sure that it is consistent.